index.js

```diff
@@ -20,35 +22,293 @@ class J5sExecutor extends Executor {
+    }
+
+    /**
+     * Jenkins job name
+     * @method _jobName
+     * @param  {String}   buildId     ID for the build
+     * @return {String}               Jenkins job name
+     */
+    _jobName(buildId) {
+        return `SD-${buildId}`;
     }
 
     /**
      * Constructor
      * @method constructor
      * @param  {Object} options           Configuration options
-     * @param  {String} options.host      Jenkins hostname to make requests to
-     * @param  {String} options.username  Jenkins username
-     * @param  {String} options.password  Jenkins password/token
+     * @param  {Object} options.ecosystem                              Screwdriver Ecosystem
+     * @param  {Object} options.ecosystem.api                          Routable URI to Screwdriver API
+     * @param  {Object} options.ecosystem.store                        Routable URI to Screwdriver Store
+     * @param  {String} options.jenkins.host                           Jenkins hostname to make requests to
+     * @param  {Number} [options.jenkins.port=8080]                    Jenkins port to make requests to
+     * @param  {String} [options.jenkins.username='screwdriver']       Jenkins username
+     * @param  {String} options.jenkins.password                       Jenkins password/token
+     * @param  {String} [options.jenkins.nodeLabel='screwdriver']      Node labels of Jenkins slaves
+     * @param  {String} [options.docker.command='docker']              THe path to the docker command
+     * @param  {String} [options.docker.launchVersion='stable']        Launcher container version to use
+     * @param  {String} [options.docker.prefix='']                     Prefix to all container names
+     * @param  {String} [options.docker.memory='4g']                   Memory limit (docker run `--memory` option)
+     * @param  {String} [options.docker.memoryLimit='6g']              Memory limit include swap (docker run `--memory-swap` option)
+     * @param  {String} [options.buildScript]                          Shell script to start the job
+     * @param  {String} [options.cleanupScript='']                     Shell script to clean up the job
+     * @param  {Number} [options.cleanupTimeLimit=20]                  Time to destory the job(seconds)
+     * @param  {Number} [options.cleanupWatchInterval=2]               Interval to detect the stopped job (seconds)
+     * @param  {String} [options.fusebox]                              Options for the circuit breaker (https://github.com/screwdriver-cd/circuit-fuses)
      */
     constructor(options) {
         super();
-        this.host = options.host;
-        this.username = options.username;
-        this.password = options.password;
+        this.ecosystem = options.ecosystem;
+        this.host = options.jenkins.host;
+        this.port = options.jenkins.port || '8080';
+        this.username = options.jenkins.username || 'screwdriver';
+        this.password = options.jenkins.password;
```

Comment:
This should probably be saved in a symbol, like secrets, to prevent accidental leakage.